client/src/views/Edit Question/EditQuestion.tsx

```diff
@@ -1,118 +1,101 @@
+                value={question.type}
+                onChange={onTypeSelected}
+                options={types} />
             <FormField
                 label="title"
                 type="text"
-                autoComplete={isActive ? "Edit title" : "Create title"}
                 value={question.title}
                 onChange={e =>
-                    SetQuestion({ ...question, title: e.target.value.trim() })
+                    setQuestion({ ...question, title: e.target.value.trim() })
                 }
                 error={titleError}
             />
 
             <FormField
                 label="aditional content"
                 type="textarea"
-                autoComplete={isActive ? "Edit content" : "Create content"}
                 value={question.additionalContent}
                 onChange={e =>
-                    SetQuestion({ ...question, additionalContent: e.target.value.trim() })
+                    setQuestion({ ...question, additionalContent: e.target.value.trim() })
                 }
                 error={titleError}
             />
-            <div>
-                <select value={questionTypes.SingleChoice} onChange={onTypeSelected}>
-                    {Object.keys(QuestionType).map((key, i) => {
-                        (
-                            <option key={key} value={key}>
-                                {QuestionType[i]}
-                            </option>
-                        )
-                    })}
-                </select>
-            </div>
-
             {/* /*not render on UI. need to check*/}
             <div> Answers
               {
-                    Array(4).map((e, i) => {
-                        (
-                            <FormField
-                                label={`${i} answer`}
-                                type="textarea"
-                                autoComplete={isActive ? "Edit answer" : "Create answer"}
-                                value={question.answers[i]?.content}
-                                onChange={e =>
-                                    SetQuestion({ ...question, answers: [{ content: e.target.value, correct: false }] }) //need to complete correct setState
-                                }
-                                error={answersError}
-                            />
-                        )
-                    })}
+                    Array(4).map(i =>
+                        <FormField
+                            label={`${i} answer`}
+                            type="textarea"
+                            value={question.answers[i]?.content}
+                            onChange={e =>
+                                setQuestion({ ...question, answers: [{ content: e.target.value, correct: false }] }) //need to complete correct setState
+                            }
+                            error={answersError}
+                        />)
+                }
```

Comment:
I am not sure I like this approach.
I think I'd rather have only one, at first, and when the last index was used add a new one. we could of course limit this to some number (maybe 10 answers max?)

This means, anytime the answers change, in the last index - if there is a value - push a new element to the index.
Then, when validating, we can allow the last one to be empty, and then simply ignore the last one.
If we apply a maximal - we have to validate this in backend as well.

Reply:
I'm awware the limitation of this aproach, I should try fix it to something more appropriate,